src/os/bluestore/BlueStore.cc

```diff
@@ -8877,84 +8877,78 @@ string BlueStore::get_device_path(unsigned id)
   return res;
 }
 
-int BlueStore::_set_bdev_label_size(const string& path, uint64_t size)
-{
-  bluestore_bdev_label_t label;
-  int r = _read_bdev_label(cct, bdev, path, &label);
-  if (r < 0) {
-    derr << "unable to read label for " << path << ": "
-          << cpp_strerror(r) << dendl;
-  } else {
-    label.size = size;
-    r = _write_bdev_label(cct, bdev, path, label);
-    if (r < 0) {
-      derr << "unable to write label for " << path << ": "
-            << cpp_strerror(r) << dendl;
-    }
-  }
-  return r;
-}
-
 int BlueStore::expand_devices(ostream& out)
 {
   int r = _open_db_and_around(true);
   ceph_assert(r == 0);
   bluefs->dump_block_extents(out);
   out << "Expanding DB/WAL..." << std::endl;
-  for (auto devid : { BlueFS::BDEV_WAL, BlueFS::BDEV_DB}) {
-    if (devid == bluefs_layout.shared_bdev ) {
+  uint64_t old_shared_size = 0;
+  uint64_t new_shared_size = 0;
+  for (auto devid : { BlueFS::BDEV_WAL, BlueFS::BDEV_DB, BlueFS::BDEV_SLOW}) {
+    bool shared_dev = devid == bluefs_layout.shared_bdev;
+    auto bdev0 = shared_dev ? bdev : bluefs->get_block_device(devid);
+    if (!bdev0 || new_shared_size) {
       continue;
     }
```

Comment:
Is this correct? If first device is shared device and it get expanded then it won't update the other devices?